docs/python-grader/index.md

```diff
@@ -179,6 +179,8 @@ The code feedback library contains built-in functions for checking correctness o
   Checks that a numpy array has the same shape and datatype as the reference solution, also checks values to see if they are close using specified `rtol` and `atol`.
 - `check_list(name, ref, data, entry_type=None)`
   Checks that a list has the same length as the reference solution. If `entry_type` is not `None`, can optionally check if each element has that type. Does _not_ check values against the reference.
+- `check_dict(name, ref, data, partial_keys=None, check_only_keys=False, check_only_values=False, entry_type_key=None, entry_type_value=None, accuracy_critical=False, report_failure=True)`
```

Comment:
In keeping with #11107 , potentially we could rename this `check_strategy="keys"|"values"|"all"`

Reply:
That's a good idea actually. Would it make sense to use a string or an enum? I think the enum is a little cleaner but not sure if it causes issues with the way the grader is laid out.

Reply:
I prefer a string with exhaustiveness checking. Enums require imports.

Reply:
I think that's reasonable for something used by client code 👍